client/src/components/molecules/ClientRequestTable.tsx

```diff
@@ -0,0 +1,154 @@
+
+interface Props {
+    requests: Request[];
+    onChangeNumRequests?: (num : number) => void;
+}
+
+const ClientRequestTable: FunctionComponent<Props> = (props: Props) => {
+    const [ requestSelectedForEditing, setRequestSelectedForEditing ] = useState("");
+    const headingList = ['Fulfilled', 'Request Group', 'Request Type', 'Quantity', 'Date Requested', ''];
+    const [requests, setRequests] = useState<Request[]>([]);
+    
+    const updateRequest = gql` 
+    mutation updateRequest($request: RequestInput){
+        updateRequest(request: $request){
+          id
+          success
+          message
+        }
+      }
+    `;
+
+    const softDeleteRequest = gql`
+    mutation deleteRequest($id: ID) {
+        softDeleteRequest(id: $id) {
+            id
+            success
+            message
+        }
+    }`
+    
+    useEffect(() => {
+        const undeletedReq : Request[] = props.requests.filter((request)=> request.deleted === false);
+        const unfulfilledReq = undeletedReq.filter(request => {
+            if (request !== undefined){
+                if (request.fulfilled === false){
+                    return request;
+                }
+            }
+        });
+        const fulfilledRequests = undeletedReq.filter(request => {
+            if (request !== undefined){
+                if (request.fulfilled === true){
+                    return request;
+                }
+            }
+        });
+        unfulfilledReq.sort((a, b)=> {
+            return (a!.dateCreated!.valueOf() - b!.dateCreated!.valueOf()); 
+        });
+        fulfilledRequests.sort((a, b)=> {
+            return (a!.dateCreated!.valueOf() - b!.dateCreated!.valueOf());
+        });
+    
+        const sortedRequests : Request[] = unfulfilledReq!.concat(fulfilledRequests!) as Request[];
+        setRequests(sortedRequests);
+      }, [props.requests]);
+      
+
+    const [mutateDeleteRequest] = useMutation(softDeleteRequest);
+    const [mutateRequest] = useMutation(updateRequest);
+    const onSoftDeleteRequest = (index: number) => {
+        const requestsCopy = requests.slice();
+        const req = {...requestsCopy[index]};
+        requestsCopy.splice(index, 1);
+        const id = req._id;
+        props.onChangeNumRequests!(requestsCopy.length);
+        setRequests(requestsCopy);
+        mutateDeleteRequest({variables: {id: id}});
+    }
+
+    const onFulfilledRequest = (index: number) => {
+        const requestsCopy = requests.slice();
+        const req = {...requestsCopy[index]};
+        if(req.fulfilled === false) { 
+            req.fulfilled = true;
+            requestsCopy.splice(index, 1) 
+            let i = requestsCopy.length - 1; 
+            for(; i > -1; --i) {
+                if(requestsCopy[i].fulfilled === false) break;
+                else if(requestsCopy[i]!.dateCreated!.valueOf() < req!.dateCreated!.valueOf()) break;
+            }
+            requestsCopy.splice(i + 1, 0, req);
```

Comment:
Readability/maintainability is not good since the intent of the code is not obvious.
Put lines 39-62 in a function and reuse the code here. The function may take longer to run, but the intent of the code will be obvious